pinot-core/src/main/java/org/apache/pinot/core/common/ObjectSerDeUtils.java

```diff
@@ -111,6 +131,19 @@ public static ObjectType getObjectType(Object value) {
         return ObjectType.Geometry;
       } else if (value instanceof RoaringBitmap) {
         return ObjectType.RoaringBitmap;
+      } else if (value instanceof LongSet) {
+        return ObjectType.LongSet;
+      } else if (value instanceof FloatSet) {
+        return ObjectType.FloatSet;
+      } else if (value instanceof DoubleSet) {
+        return ObjectType.DoubleSet;
+      } else if (value instanceof ObjectSet) {
+        ObjectSet objectSet = (ObjectSet) value;
+        if (objectSet.isEmpty() || objectSet.iterator().next() instanceof String) {
```

Comment:
will this be a problem as we always return StringSet for empty value?

Reply:
It is fine for empty set as empty string set and empty bytes set are the same (both are empty ObjectOpenHashSet)